CONTRIBUTING.md

```diff
@@ -53,10 +54,96 @@ $ cp config/application.example.yml config/application.yml
     local database. If your local postgres user has a different password, make
     sure to change that in the `development` and `test` sections of
     `database.yml`
+    It's important that host is set to localhost in `database.yml`
 8.  Set up the database:
 ```
 $ bundle exec rake db:test:prepare
 ```
+
+If this step fails on Ubuntu, make sure that your postgres db is up and running 
+and that you only have one postgres instance up: try `sudo lsof -i:5432`. 
+You may need to use sudo to set your `/etc/postgresql/11/main/pg_hba.conf` file. 
+IPv4 local connections should say `localhost` under ADDRESS and `trust` under METHOD.
+9.  Now you should be able to run tests locally, and they should all pass:
+```
+$ bundle exec rake spec
+```
+10. Optionally, if you want to run the doorbell code locally, you will need to have a local instance of [Redis](https://redis.io/). You can install it on mac with `brew install redis`. If your local Redis instance is running on a non-standard port, you can set `REDIS_URL` in your local `config/application.yaml`.
+
+### Steps to run arooo server locally
+
+1. `bundle exec rake populate:users` to set up dummy data
+1. `bundle exec rails server` don't forget to use http://localhost:3000 and not https
+1. `bundle exec rails console` Optional - useful for looking at and changing your local data)
+
+### Docker setup (optional)
+
+1. Install docker and docker compose# Contributing Guide
```

Comment:
Perhaps you accidentally copy+pasted the entire document back into itself at this point? It seems most of it duplicated starting from here on down.

Reply:
Hmm, I didn't think I copy-pasted anything. I deleted a line and this is how Git decided to handle the deleted line I think.

Reply:
Oh no wait, you're right there is a duplicate. I should fix this.